modules/library/render/src/main/java/org/geotools/renderer/style/SLDStyleFactory.java

```diff
@@ -983,6 +987,12 @@ private Geometry findGeometry(final Object feature, String geomName) {
         if(kerning) {
             javaFont = applyKerning(javaFont);
         }
+        double spacing = voParser.getDoubleOption(symbolizer, TextSymbolizer.CHAR_SPACING_KEY, 
+                TextSymbolizer.DEFAULT_CHAR_SPACING);
+        if(spacing != 0) {
```

Comment:
Is this just a common-case optimisation? It should still work with spacing == 0?